File: frontend/src/types/workflow.ts
export interface N8nNode {
  id?: string;
  name: string;
  type: string;
  position: number[];
  parameters: Record<string, unknown>;
  typeVersion?: number;
  credentials?: Record<string, unknown>;
}

export interface N8nWorkflow {
  name: string;
  nodes: N8nNode[];
  connections: Record<string, unknown>;
  settings?: Record<string, unknown>;
  active?: boolean;
}

File: backend/models/workflow.py
from pydantic import BaseModel
from typing import Optional, Any


class N8nNode(BaseModel):
    id: Optional[str] = None
    name: str
    type: str
    position: list[int]
    parameters: dict = {}
    typeVersion: Optional[float] = None


class N8nConnection(BaseModel):
    node: str
    type: str = "main"
    index: int = 0


class N8nWorkflow(BaseModel):
    name: str = "Generated Workflow"
    nodes: list[dict]
    connections: dict
    settings: Optional[dict] = None
    active: bool = False


class N8nExecution(BaseModel):
    id: str
    finished: bool
    mode: str
    status: Optional[str] = None
    data: Optional[dict] = None


class N8nWorkflowResponse(BaseModel):
    id: str
    name: str
    active: bool
    nodes: list[dict]
    connections: dict
    createdAt: Optional[str] = None
    updatedAt: Optional[str] = None
